engine/chassis.ts

```diff
@@ -478,7 +486,10 @@ export class Program {
             bindings.descriptorSets.map((descriptorSet, index) => commandBuffer.bindDescriptorSet(index, descriptorSet));
         } else {
             for (let i = 0; i < bindings.descriptorSets.length; i++) {
-                const offsets = bindings.dynamicOffsets[i].map((offset) => offset * instance);
+                const offsets = bindings.dynamicOffsets[i];
+                for (let j = 0; j < offsets.length; j++) {
+                    offsets[j] *= instance;
```

Comment:
Remember the assignment operator is just taking references, not copy by value.

So I think this accumulates the same offsets every time the `*=` is invoked. Not sure if this works for more than 1 instances.